graphql_compiler/schema_transformation/rename_query.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+from copy import deepcopy
+
+from graphql.language import ast as ast_types
+from graphql.language.visitor import Visitor, visit
+
+from .utils import QueryStructureError
+
+
+def rename_query(ast, renamings):
+    """Translate all types and root fields (fields of query type) using renamings.
+
+    Most fields (fields of types other than the query type) will not be renamed. Any type
+    or field that does not appear in renamings will be unchanged.
+    Roughly speaking, rename_query renames the same elements as rename_schema.
+
+    Args:
+        ast: Document, representing a valid query. It is assumed to have passed GraphQL's
+             builtin validation -- validate(schema, ast). ast not modified by this function
+        renamings: Dict[str, str], mapping original types/query type field names as appearing
+                   in the query to renamed names. Type or query type field names not appearing
+                   in the dict will be unchanged
+
+    Returns:
+        Document, a new AST representing the renamed query
+
+    Raises:
+        - QueryStrutureError if the ast does not have the expected form; in particular, if the
+          AST contains Fragments, or if it contains an InlineFragment at the root level
+    """
+    # NOTE: There is a validation section in graphql-core that takes in a schema and a
+    # query ast, and checks whether the query is valid. This code assumes this validation
+    # step has been done on the input AST.
+    if len(ast.definitions) > 1:  # includes either multiple queries, or fragment definitions
+        raise QueryStructureError(
```

Comment:
There are already exceptions regarding query validity in graphql_compiler/exceptions so I think it's best you either use one from there or put `QueryStructureError` there in order to have all exceptions of the same kind in one place.

Reply:
I took a look at those errors and feel uncertain of what I should do. compiler_frontend.py checks for some of the same criteria that I do here (for example, not starting with type coercion) and raise GraphQLCompilationError if the check fails, but I'm not doing compilation here, just a transformation. So that error doesn't feel right even if it's raised under some of the same conditions. The other existing errors don't apply either. What are your thoughts?